File: 0912-sort-an-array/0912-sort-an-array.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        if len(nums) == 0 or max(nums) == min(nums):
            return nums
        mid = (max(nums) + min(nums)) / 2
        smallers, biggers = [], []

        for num in nums:
            if num < mid:
                smallers.append(num)
            else:
                biggers.append(num)

        return self.sortArray(smallers) + self.sortArray(biggers)

File: 0912-sort-an-array/0912-sort-an-array.cpp
class Solution {
public:
    vector<int> sortArray(vector<int>& nums) {
        if(nums.empty() || *min_element(nums.begin(), nums.end()) == *max_element(nums.begin(), nums.end())) {
            return nums;
        }
        vector<int> smallers, biggers;
        float mid = (*min_element(nums.begin(), nums.end()) + *max_element(nums.begin(), nums.end())) / 2.0;
        
        for(auto& num : nums) {
            if(num < mid) {
                smallers.push_back(num);
            } else {
                biggers.push_back(num);                
            }
        }
        
        smallers = sortArray(smallers);
        biggers = sortArray(biggers);
        smallers.insert(smallers.end(), biggers.begin(), biggers.end());
        return smallers;
    }
};